example/next-storage/pages/index.tsx

```diff
@@ -0,0 +1,80 @@
+import { useState, useEffect } from 'react'
+import { supabase } from '../lib/api'
+import Auth from '../components/Auth'
+import Account from '../components/Account'
+import ProfileCard from '../components/ProfileCard'
+import { AuthSession, Subscription, SupabaseRealtimePayload } from '../../../dist/main'
+import { Profile } from '../lib/constants'
+
+var realtimeProfiles: Subscription | null
+
+export default function Home() {
+  const [session, setSession] = useState<AuthSession | null>(null)
+  const [profiles, setProfiles] = useState<Profile[]>([])
+
+  useEffect(() => {
+    setSession(supabase.auth.session())
+    getPublicProfiles()
+
+    supabase.auth.onAuthStateChange((_event: string, session: AuthSession | null) => {
+      setSession(session)
+    })
+
+    realtimeProfiles = supabase
```

Comment:
@phamhieu you had a problem with realtime only returning one item right? What code changes did you make to fix it? I want to make sure we're suggesting the best method for next.js/react

Reply:
https://github.com/phamhieu/realtime-map/commit/f2d50019e21588c30405243b853d73e45717e72d

I update the code to use `useReducer`. Then on realtime callback i use `dispatch` to trigger change on my UI. It works nicely

Reply:
Nice 🔥